pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -70,7 +78,25 @@ type datasourceService struct {
 	HTTPClient *http.Client
 }
 
-func NewInstanceSettings(cfg *setting.Cfg) datasource.InstanceFactoryFunc {
+func getDatasourceService(cfg *setting.Cfg, clientProvider httpclient.Provider, dsInfo datasourceInfo, routeName string) (datasourceService, error) {
+	srv, ok := dsInfo.Services[routeName]
+	if ok && srv.HTTPClient != nil {
```

Comment:
If this happens seems like a bug to me since `getDatasourceService` is only called from `NewInstanceSettings`. Maybe more appropriate to return an error?